### PR TITLE
Major refactor in TodoList examples

### PR DESCRIPTION
**Pull request by JesusMtnez**
This PR migrates TodoList examples to tagless algebras and adds an example using `http4s`

### CODE REVIEW

File: modules/examples/todolist-http-finch/src/main/scala/todo/TodoListApp.scala [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      handler: F ~> Future,
+      T: Transactor[F],
+      api: Api[F]): F[ListeningServer] = {
+
+    val service: Service[Request, Response] = api.endpoints.toService
+    val log: LoggingM[F]                    = app.services.log
+    val cfg: ConfigM[F]                     = app.services.config
+
+    for {
+      _      <- log.warn("Trying to load application.conf")
```
Comment (AdrianRaFo) [inline modules/examples/todolist-http-finch/src/main/scala/todo/TodoListApp.scala]: An Info level maybe is more adequated here

**Review by juanpedromoreno — APPROVED** [review]
:shipit:

File: modules/examples/todolist-http-finch/src/main/scala/todo/TodoListApp.scala (hunk 1 above)
Comment (JesusMtnez) [reply modules/examples/todolist-http-finch/src/main/scala/todo/TodoListApp.scala]: Done!

**Review by fedefernandez — APPROVED** [review]
👏

**Pull request merged**